### PR TITLE
[draft] Drift consolidation controller

### PR DESCRIPTION
**Pull request by ra-grover**
Fixes # 

**Description**
The PR scaffolds Drift consolidation controller.
It's related to https://github.com/aws/karpenter/pull/2812

**How was this change tested?**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/provisioning/v1alpha5/labels.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var (
 	TerminationFinalizer              = Group + "/termination"
 	LabelNodeInitialized              = Group + "/initialized"
 	LabelCapacityType                 = Group + "/capacity-type"
+	DriftedAnnotationKey              =	Group + "/drifted-node"
```
Comment (ellistarn) [inline pkg/apis/provisioning/v1alpha5/labels.go:41]: I'd love to abstract this concept into `karpenter.sh/voluntary-disruption = drifted`. @jonathan-innis can you chime in? This enables us to write additional integrations for other case like node rebalance, etc. 

```
karpenter.sh/voluntary-disruption = drifted
karpenter.sh/voluntary-disruption = rebalance
karpenter.sh/voluntary-disruption = expired
```

Comment (ra-grover) [reply pkg/apis/provisioning/v1alpha5/labels.go:41]: Happy to do this way, I think @njtran in https://github.com/aws/karpenter/pull/2812 mentioned to use this as a label. Let me know and I would convert this to a label as you mentioned.

Comment (ellistarn) [reply pkg/apis/provisioning/v1alpha5/labels.go:41]: He and I are on the same page. Happy to accept as you have it, and we'll tidy it up later. @jonathan-innis FYI.

**Comment by njtran** [comment]
Hey @ra-grover, closing this as we discussed.

**Pull request closed without merging**